model/src/main/java/uk/gov/pay/commons/api/validation/ExternalJsonMetadataValidator.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.pay.commons.api.validation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Iterator;
+import java.util.Map;
+
+public class ExternalJsonMetadataValidator implements ConstraintValidator<ValidExternalJsonMetadata, JsonNode> {
+
+    private static final int MAXIMUM_KEYS = 10;
+    private static final int MAXIMUM_VALUE_LENGTH = 50;
+    private static final int MAXIMUM_KEY_LENGTH = 30;
+
+    @Override
+    public boolean isValid(JsonNode jsonNode, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
I think the readability of this method could be improved by splitting it out into functions, e.g.
```
public boolean isValid(JsonNode jsonNode, ConstraintValidatorContext constraintValidatorContext) {
        if (jsonNode == null) {
            return true;
        }
     
        constraintValidatorContext.disableDefaultConstraintViolation();

        validateKeyValuePairs(jsonNode);
        validateKeysAreNotNull(List<String>);
        validateKeyLengths(List<String>);
        validateValues(Iterator<Map.Entry<String, JsonNode>>);
}
```

Reply:
something like that

Reply:
yeah thanks for the feedback Oz. You're right it could be made clearer. It can't be decomposed quite like the above since we need to validate keys and pairs together for many of the conditions so that we can reference them together in the error messages. I've refactored it and force pushing with another go. Happy to hear your thoughts again.

Reply:
how about I approve this PR so you can carry on with the rest story and i'll make another PR with the refactor I have in mind? then you can see if it makes it easier or not :)

Reply:
yes sounds good, thanks Oz.